pupil_src/shared_modules/video_capture/utils.py

```diff
@@ -526,3 +529,10 @@ def load_worn_data(path):
             conf_data = (1.0 for _ in range(len(timestamps)))
 
         yield from zip(raw_data, timestamps, conf_data)
+
+
+def match_contents_by_name_pattern(parent_dir: Path, name_pattern) -> T.List[Path]:
+    # Get all non-recursive directory contents
+    contents = parent_dir.absolute().glob("*")
+    # Filter content that matches the name by regex pattern
+    return [c for c in contents if re.match(name_pattern, c.name) is not None]
```

Comment:
I think the function name could be better, `match` sounds like it returns a boolean or some match-object.
Maybe `filter_dir`? This would be similar to `os.listdir()` which also does not go into sub-directories.
Also, the type `str` of name_pattern is missing.

Proposals:
A) Simple, maybe not clear if filtering name or full path?
```python
def filter_dir(dir: Path, name_pattern: str) -> T.List[Path]:
```
B) More verbose about what is filtered
```python
def filter_dir_by_name_pattern(dir: Path, name_pattern: str) -> T.List[Path]:
```

Reply:
I would suggest `matched_files_from_pattern` and using `contents = filter(Path.is_file, parent_dir.iterdir())`